internal/app/referenceserver/server.go

```diff
@@ -191,9 +194,6 @@ func createServer(req *v1alpha1.ServerCompatRequest, listenAddr string) (httpSer
 
 // newH1Server creates a new HTTP/1.1 server.
 func newH1Server(handler http.Handler, listenAddr string, tlsConf *tls.Config) (httpServer, error) {
-	if tlsConf != nil {
-		tlsConf.NextProtos = []string{"http/1.1"}
-	}
```

Comment:
This was basically a no-op, so I removed it. It's the _client_ that proposes the protocols, so this was important to set on the client to make sure that HTTP/1.1 or HTTP/2 gets used. It's not necessary on the server.